packages/@uppy/core/src/index.js

```diff
@@ -539,6 +557,33 @@ class Uppy {
     }
   }
 
+  /**
+   * Check if requiredMetaField restriction is met before uploading.
+   *
+   * @private
+   */
+  checkRequiredMetaFields (files) {
+    const { requiredMetaFields } = this.opts.restrictions
+    const { hasOwnProperty } = Object.prototype.hasOwnProperty
+
+    const errors = []
+    const fileIDs = Object.keys(files)
+    for (let i = 0; i < fileIDs.length; i++) {
```

Comment:
I wonder if `forEach` would be clearer here? Nested `i`s are a bit overcomplicating it, I think.

Reply:
Is it OK to leave it at that for now and move that to a `for of` loop in `v2.0.0`?